Rubberduck.Parsing/VBA/RubberduckParser.cs

```diff
@@ -206,18 +202,65 @@ private void ParseAll()
                 }
             }
 
-            var parseTasks = new Task[components.Count];
-            for (var i = 0; i < components.Count; i++)
+            _projectDeclarations.Clear();
+            _state.ClearBuiltInReferences();
+
+            var parseTasks = new Task[toParse.Count];
+            for (var i = 0; i < toParse.Count; i++)
             {
-                parseTasks[i] = ParseAsync(components[i], CancellationToken.None);
+                var index = i;
+                parseTasks[i] = new Task(() =>
+                {
+                    ParseAsync(toParse[index], CancellationToken.None).Wait();
```

Comment:
If we always wait after invoking Parse (as in wrap calls to the parser in separate Tasks anyways) then we could just make the parser calls synchronous.